### PR TITLE
Improve modify functions and test coverage

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

See #303 for more information ...

**Why**:

GSA, etc. will need this ...

**How**:

We introduce the `UNDEFINED_VALUE`, that is different to `None`.
This way we can differ if the value is set to `None` or not set at all.

**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

**Comment by y0urself** [comment]
⚠️ ⚠️ ⚠️  Attention. Merging this has a huge impact for Hyperion. ⚠️ ⚠️ ⚠️

File: gvm/protocols/base.py [hunk 1]
```diff
@@ -35,6 +39,8 @@ class GvmProtocol:
             See :mod:`gvm.transforms` for existing transforms.
     """
 
+    _UNDEFINED_VALUE = UndefinedValue()
```
Comment (bjoernricks) [inline gvm/protocols/base.py:42]: Personally I would change this into a module variable instead of a class variable but this is more or less just a matter of taste.

**Comment by y0urself** [comment]
Do we merge this one?

File: gvm/protocols/gmpv8/gmpv8.py [hunk 2]
```diff
@@ -925,55 +925,6 @@ def create_filter(
 
         return self._send_xml_command(cmd)
 
-    def modify_filter(
```
Comment (bjoernricks) [inline gvm/protocols/gmpv8/gmpv8.py:928]: The change should have been done in another PR. It is very difficult to review PRs that are getting bigger and mixing several purposes.

File: gvm/protocols/gmpv8/gmpv8.py [hunk 3]
```diff
@@ -538,14 +537,16 @@ def modify_tag(
...
-        if comment:
-            cmd.add_element("comment", comment)
+        self._check_input('comment', comment, cmd)
 
         if name:
             cmd.add_element("name", name)
 
-        if value:
-            cmd.add_element("value", value)
+        if value != GvmProtocol._UNDEFINED_VALUE:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv8/gmpv8.py:545]: Isn't this something for `_check_input` too?

File: gvm/protocols/gmpv8/gmpv8.py [hunk 4]
```diff
@@ -1089,16 +1040,18 @@ def modify_schedule(
         cmd = XmlCommand("modify_schedule")
         cmd.set_attribute("schedule_id", schedule_id)
 
-        if name:
-            cmd.add_element("name", name)
+        if name != GvmProtocol._UNDEFINED_VALUE:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv8/gmpv8.py:1043]: Same. Isn't this something for `_check_input`?

File: gvm/protocols/gmpv9/gmpv9.py [hunk 5]
```diff
@@ -53,23 +53,7 @@ def _check_event(
...
-                "method is required for event {}".format(event.name)
-            )
-
-        if condition not in (
-            AlertCondition.ALWAYS,
-            AlertCondition.FILTER_COUNT_CHANGED,
-            AlertCondition.FILTER_COUNT_AT_LEAST,
-            AlertCondition.SEVERITY_AT_LEAST,
-            AlertCondition.SEVERITY_CHANGED,
-        ):
```
Comment (bjoernricks) [inline gvm/protocols/gmpv9/gmpv9.py:72]: I really don't know why this method has been changed. Should have been put into another PR.

File: gvm/protocols/gmpv7/gmpv7.py [hunk 6]
```diff
@@ -102,57 +100,10 @@ def _is_list_like(value: Any) -> bool:
 def _check_event(
     event: AlertEvent, condition: AlertCondition, method: AlertMethod
 ):
-    if event == AlertEvent.TASK_RUN_STATUS_CHANGED:
```
Comment (bjoernricks) [inline gvm/protocols/gmpv7/gmpv7.py:105]: I am really not sure about changing `_check_event` as I don't understand the changes at the moment.

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Could you check the usage of `_check_input`. For me it seems there are still a lot of places where it could be used.

This PR has become a bit difficult to read and review. It contains too many commits already and too many unrelated changes. I know sometimes it looks like much overhead for creating another PR for simple changes but it is way easier to review and therefore it will get merged much faster.

File: gvm/protocols/gmpv8/gmpv8.py (hunk 4 above)
Comment (y0urself) [reply gvm/protocols/gmpv8/gmpv8.py:1043]: I will create some other PRs and split things up ...

**Comment by y0urself** [comment]
Will rethink this issue.
